portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +35,32 @@
 public final class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
-
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    String filter = request.getParameter("filter");
```

Comment:
You can't rely on the client setting the "filter" parameter ("the client can't be trusted"). I recommend checking the input very closely here at the beginning, and e.g. fail appropriately if you decide that the "filter" parameter is mandatory.